File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
        String result = "";
        while (n != 0) {
            result += String.valueOf(n % 3);
            n /= 3;
        }
        
        int answer = 0;
        int length = result.length(); int num = 0;
        for (int i = 0; i < length; i++) {
            num = result.charAt(length - i - 1) - '0';
            answer += (int) Math.pow(3, i) * num;
        }
        return answer;
    }
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n):
    result = []
    while n != 0:
        div, mod = divmod(n, 3)
        result.append(mod)
        n //= 3
    
    answer = 0
    result.reverse()
    for i in range(len(result)):
        answer += (3 ** i) * result[i]
    return answer